File: python/high-scores/high_scores.py
class HighScores:
    def __init__(self, scores):
        self.scores = scores

    def latest(self):
        return self.scores[-1]

    def personal_best(self):
        return max(self.scores)

    def personal_top_three(self):
        copy = [*self.scores]
        best = []
        for i in range(3):
            if not len(copy): break
            m = max(copy)
            del copy[copy.index(m)]
            best.append(m)
        return best

File: ruby/high-scores/high_scores.rb
class HighScores
  def initialize(scores)
    @scores = scores
  end

  def scores
    @scores
  end

  def latest
    @scores[-1]
  end

  def personal_best
    @scores.max
  end

  def latest_is_personal_best?
    self.personal_best == self.latest
  end

  def personal_top_three
    s = @scores.clone
    three = []
    (1..3).each {
      if s.empty? then next end
      three << s.max
      s.delete_at(s.index(s.max))
    }
    return three
  end


end
